mcs/class/referencesource/mscorlib/system/collections/generic/list.cs

```diff
@@ -997,8 +997,7 @@ public void Sort(IComparer<T> comparer)
             Contract.EndContractBlock();
 
             if( _size > 0) {
-                IComparer<T> comparer = new Array.FunctorComparer<T>(comparison);
-                Array.Sort(_items, 0, _size, comparer);
+                ArraySortHelper<T>.Sort(_items, 0, _size, comparison);
```

Comment:
could you please do this in an `#if MONO` block? it makes merging upstream referencesource changes easier :)